src/api/spec/models/user_spec.rb

```diff
@@ -623,4 +623,43 @@
       end
     end
   end
+
+  describe '#can_modify?' do
+    let(:user) { create(:confirmed_user, :with_home, login: 'hans') }
+    let(:project) { user.home_project }
+
+    context 'projects' do
+      context 'can modify a home project' do
+        it { expect(user.can_modify?(project)).to be true }
+      end
+
+      context 'can modify sub-projects of a home project' do
+        let(:child_project) { create(:project, name: "#{project.name}:something") }
+
+        it { expect(user.can_modify?(child_project)).to be true }
+      end
+
+      context 'when the project has no parent project' do
+        let(:user) { create(:confirmed_user) }
+
+        context 'but the user is the root owner' do
+          let(:child_project) { create(:project, name: "home:#{user.login}:something") }
+
+          it { expect(user.can_modify?(child_project)).to be true }
+        end
+      end
+
+      context "cannot modify other user's home projects" do
+        let(:other_user) { create(:confirmed_user, :with_home, login: 'hans2') }
+
+        it { expect(user.can_modify?(other_user.home_project)).to be false }
+      end
+
+      context "cannot modify other people's projects" do
```

Comment:
What's the difference to the spec above?

Reply:
None, I'll use this context as it's more generic.